src/modules/campaign/ContentfulClient.spec.ts

```diff
@@ -1,87 +1,194 @@
+import { ContentfulLocale } from '@dcl/schemas'
+
+const CMS_URL = 'https://cms.decentraland.org'
 
 describe('ContentfulClient', () => {
   let client: ContentfulClient
+  const mockSpace = 'test-space'
+  const mockEnvironment = 'test-env'
+  const mockId = 'test-id'
 
   beforeEach(() => {
     client = new ContentfulClient()
+    nock.cleanAll()
+  })
+
+  afterEach(() => {
+    nock.cleanAll()
   })
 
   describe('fetchEntry', () => {
-    const mockSpace = 'test-space'
-    const mockEnvironment = 'test-env'
-    const mockId = 'test-id'
-    const mockContentType = 'test-content-type'
-    const mockToken = 'test-token'
-
-    describe('when the request is successful', () => {
-      let mockResponse: ContentfulResponse<any>
+    describe('and the request is successful', () => {
       beforeEach(() => {
-        mockResponse = {
-          items: [{ ...mockHomepageBannerEntry }],
-          includes: {
-            Asset: [],
-            Entry: []
-          }
-        }
-        global.fetch = jest.fn().mockResolvedValue({
+        nock(CMS_URL)
+          .get('/spaces/space-id/environments/env-id/entries/entry-id/')
+          .query({ locale: 'en-US' })
+          .reply(200, mockAdminEntry)
+      })
+
+      it('should return the entry data', async () => {
+        const result = await client.fetchEntry('space-id', 'env-id', 'entry-id')
+        expect(result).toEqual(mockAdminEntry)
+      })
+    })
+
+    describe('and the request fails', () => {
+      beforeEach(() => {
+        nock(CMS_URL)
+          .get('/spaces/space-id/environments/env-id/entries/entry-id/')
+          .query({ locale: 'en-US' })
+          .reply(500)
+      })
+
+      it('should throw an error', async () => {
+        await expect(
+          client.fetchEntry('space-id', 'env-id', 'entry-id')
+        ).rejects.toThrow('Failed to fetch entity data')
+      })
+    })
+  })
+
+  describe('fetchEntryAllLocales', () => {
+    it('should combine fields from all available locales', async () => {
+      const mockResponses = {
+        enUS: {
+          fields: { title: { [ContentfulLocale.enUS]: 'English Title' } }
+        },
+        es: {
+          fields: { title: { [ContentfulLocale.es]: 'Título en Español' } }
+        },
+        zh: { fields: { title: { [ContentfulLocale.zh]: '中文标题' } } }
+      }
+
+      global.fetch = jest
+        .fn()
+        .mockResolvedValueOnce({
           ok: true,
-          json: () => Promise.resolve(mockResponse)
+          json: () => Promise.resolve(mockResponses.enUS)
         })
+        .mockResolvedValueOnce({
+          ok: true,
+          json: () => Promise.resolve(mockResponses.es)
+        })
+        .mockResolvedValueOnce({
+          ok: true,
+          json: () => Promise.resolve(mockResponses.zh)
+        })
+
+      const result = await client.fetchEntryAllLocales(
+        mockSpace,
+        mockEnvironment,
+        mockId
+      )
+
+      expect(result).toEqual({
```

Comment:
We're generally missing the cases where the requests fail, would you mind adding tests for these cases?